stores/sqlstore.go

```diff
@@ -1664,8 +1699,80 @@ func (ms *SQLMsgStore) flush() error {
 		return err
 	}
 	tx = nil
-	if ms.limits.MaxAge > 0 && ms.expireTimer == nil {
-		ms.createExpireTimer()
+	return nil
+}
+
+// Insert messages with INSERT INTO MESSAGES () VALUES (),(),()...
+// This is faster than the original insert with transactions.
+// It is done only if user configures the BulkInsertLimit option.
+// Lock held on entry.
+func (ms *SQLMsgStore) bulkInsert(limit int) error {
+	const insertStmt = "INSERT INTO Messages (id, seq, timestamp, size, data) VALUES "
```

Comment:
why two different strings for the template?

Reply:
Sorry, I am not sure I understand. There is only 1 "insert" statement value, which is this one. The values however are different because mysql wants: "(?,?,?,?,?)", while postgres wants: "($1, $2, $3, $4, $5)".